lib/storage-handler.js

```diff
@@ -11,9 +11,29 @@ var IncomingForm = require('formidable');
 var StringDecoder = require('string_decoder').StringDecoder;
 var path = require('path');
 var uuid = require('uuid');
+var fs = require('fs');
+var defaultOptions = function() {
+  var dataSources = path.join(__dirname, '../../../server/datasources.json');
```

Comment:
The datasource settings should be passed in from the connector. Importing a relative file is not reliable.

Reply:
path.join() do the job of get the absolute path.

Reply:
It's not about the absolute path. Please note the `loopback-component-storage` module can be installed somewhere related to the loopback application. Your assumption is not always true.

Reply:
In this case, `StorageService` class in `lib/storage-service.js` has the data source settings. But `lib/storage-handler` is a singleton. You should convert it a class so that it receive `settings` too.